packages/template-retail-react-app/app/components/product-view/index.jsx

```diff
@@ -220,16 +222,17 @@ const ProductView = forwardRef(
                     <Button
                         key="cart-button"
                         onClick={handleCartItem}
-                        disabled={showInventoryMessage}
+                        disabled={isBasketLoading || showInventoryMessage}
+                        isLoading={isBasketLoading}
```

Comment:
This is the other main part of the fix. This disables the add to cart button and shows a spinner until isBasketLoading (set by the PDP) becomes false

Reply:
Add some tests to validate the new loading state?

Reply:
Unit tests have been added!